zhaquirks/tuya/ts1201.py

```diff
@@ -0,0 +1,524 @@
+"""Tuya TS1201 IR blaster.
+
+   Heavily inspired by work from @mak-42
+https://github.com/Koenkk/zigbee-herdsman-converters/blob/9d5e7b902479582581615cbfac3148d66d4c675c/lib/zosung.js
+"""
+import base64
+from typing import Any, List, Optional, Union
+
+from zigpy.profiles import zha
+from zigpy.quirks import CustomCluster, CustomDevice
+import zigpy.types as t
+from zigpy.zcl import foundation
+from zigpy.zcl.clusters.general import (
+    Basic,
+    GreenPowerProxy,
+    Groups,
+    Identify,
+    OnOff,
+    Ota,
+    PowerConfiguration,
+    Scenes,
+    Time,
+)
+
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+
+
+class Bytes(bytes):
+    """Bytes serializable class."""
+
+    def serialize(self):
+        """Serialize Bytes."""
+        return self
+
+    @classmethod
+    def deserialize(cls, data):
+        """Deserialize Bytes."""
+        return cls(data), b""
+
+
+class ZosungIRControl(CustomCluster):
+    """Zosung IR Control Cluster (0xE004)."""
+
+    name = "Zosung IR Control Cluster"
+    cluster_id = 0xE004
+    ep_attribute = "zosung_ircontrol"
+
+    attributes = {0x0000: ("last_learned_ir_code", t.CharacterString, True)}
+
+    server_commands = {
+        0x00: foundation.ZCLCommandDef(
+            "data",
+            schema={"data": Bytes},
+            direction=foundation.Direction.Server_to_Client,
+            is_manufacturer_specific=True,
+        ),
+        0x01: foundation.ZCLCommandDef(
+            "IRLearn",
+            schema={"on_off": t.Bool},
+            direction=foundation.Direction.Server_to_Client,
+            is_manufacturer_specific=True,
+        ),
+        0x02: foundation.ZCLCommandDef(
+            "IRSend",
+            schema={"code": t.CharacterString},
+            direction=foundation.Direction.Server_to_Client,
+            is_manufacturer_specific=True,
+        ),
+    }
```

Comment:
Ideally, this would be switched to the new zigpy `AttributeDefs` and `ServerCommandDefs` (similar to [this example](https://github.com/zigpy/zigpy/blob/4570a0a47b6f2ad56f922f8cd5861bbfdda25915/zigpy/zcl/clusters/homeautomation.py#L524-L534)).